### PR TITLE
Updates to work with newer libraries

### PR DESCRIPTION
**Pull request by joshua-cogliati-inl**
--------
Pull Request Description
--------
##### What issue does this change request address? 
#224 

##### What are the significant changes in functionality due to this change request?
Removes the requirement that sklearn be 0.18, since it now works with 0.19

Note that this should not be squashed when merging since there were three separate changes needed.

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [ ] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

**Comment by PaulTalbot-INL** [comment]
Starting a review.

File: framework/PostProcessors/BasicStatistics.py [hunk 1]
```diff
@@ -935,7 +935,8 @@ def startVector(metric):
         paramSamples[p,:] = input['targets'][param][:]
         pbWeightsList[p] = pbWeights['realization'] if param not in pbWeights['SampledVarsPbWeight']['SampledVarsPbWeight'].keys() else pbWeights['SampledVarsPbWeight']['SampledVarsPbWeight'][param]
       pbWeightsList.append(pbWeights['realization'])
-      if None in pbWeightsList:
+      #if None in pbWeightsList:
```
Comment (PaulTalbot-INL) [inline framework/PostProcessors/BasicStatistics.py]: Do we want to leave this comment for any particular reason?

File: tests/framework/tests [hunk 2]
```diff
@@ -291,6 +291,7 @@
    xml = 'test_iostep_load/reprint_HistorySet.xml'
    prereq = test_output
    remove_unicode_identifier = true
+   remove_whitespace = true
```
Comment (PaulTalbot-INL) [inline tests/framework/tests]: Why is this needed for this test in particular?

File: framework/PostProcessors/BasicStatistics.py (hunk 1 above)
Comment (joshua-cogliati-inl) [inline framework/PostProcessors/BasicStatistics.py]: Well, it is more understandable than what I changed it into, but maybe I should turn it into a proper comment.

File: tests/framework/tests (hunk 2 above)
Comment (joshua-cogliati-inl) [inline tests/framework/tests]: "'64'], \n" in the gold versus "'64'],\n" in the new version.

File: framework/PostProcessors/BasicStatistics.py (hunk 1 above)
Comment (PaulTalbot-INL) [inline framework/PostProcessors/BasicStatistics.py]: Ok.  I think we can remove the comment; I don't think the replacement is too obscure.  Whatever you like best, though.

File: tests/framework/tests (hunk 2 above)
Comment (PaulTalbot-INL) [inline tests/framework/tests]: Okay, thanks.

File: framework/PostProcessors/BasicStatistics.py (hunk 1 above)
Comment (joshua-cogliati-inl) [inline framework/PostProcessors/BasicStatistics.py]: Fixed.

**Comment by joshua-cogliati-inl** [comment]
Failure:
```
\tests\framework.test_comparison_statistics: (  130.81 sec) STEP IOCOMBINED          : Message         -> ***       Run finished        ***
\tests\framework.test_comparison_statistics: (  130.81 sec) STEP IOCOMBINED          : Message         -> ***     Closing the step      ***
\tests\framework.test_comparison_statistics: (  130.81 sec) STEP IOCOMBINED          : Message         -> ***        Step closed        ***
\tests\framework.test_comparison_statistics: (  130.81 sec) SIMULATION               : Message         -> -- End step dump_data of type: IOStep                          --
\tests\framework.test_comparison_statistics: 
\tests\framework.test_comparison_statistics: (  130.81 sec) SIMULATION               : Message         -> Run complete!
\tests\framework.test_comparison_statistics: --------------------------------------------------
\tests\framework.test_comparison_statistics: There were 1 warnings during the simulation run:
\tests\framework.test_comparison_statistics: (1 time) cpu_count 2 < batchSize 3
\tests\framework.test_comparison_statistics: --------------------------------------------------
\tests\framework.test_comparison_statistics: 
\tests\framework.test_comparison_statistics: ################################################################################
\tests\framework.test_comparison_statistics: Process terminated by test harness. Max time exceeded (300 seconds)
\tests\framework.test_comparison_statistics: ################################################################################
\tests\framework.test_comparison_statistics: 
```

File: framework/PostProcessors/BasicStatistics.py (hunk 1 above)
Comment (PaulTalbot-INL) [inline framework/PostProcessors/BasicStatistics.py]: I don't see the commit where this was pushed.  Am I missing it?

Comment (joshua-cogliati-inl) [inline framework/PostProcessors/BasicStatistics.py]: I force push a new version of history, where I never commented out ```if None in pbWeightsList:```

**Comment by PaulTalbot-INL** [comment]
Approved to merge if tests pass without additional code changes.

**Comment by joshua-cogliati-inl** [comment]
All the tests passed (including the one that github doesn't think passed.)
